File: AdminArea/AdminArea.Web/Modules/ServerTypes/ProductImagescom/ProductImagescomColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductImagescomRow } from "./ProductImagescomRow";

export interface ProductImagescomColumns {
    Id: Column<ProductImagescomRow>;
    ProductName: Column<ProductImagescomRow>;
    Image: Column<ProductImagescomRow>;
}

export class ProductImagescomColumns extends ColumnsBase<ProductImagescomRow> {
    static readonly columnsKey = 'ProductImagescom.ProductImagescom';
    static readonly Fields = fieldsProxy<ProductImagescomColumns>();
}

File: AdminArea/AdminArea.Web/Modules/ProductImagescom/ProductImagescom/ProductImagescomColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminArea.ProductImagescom.Columns;

[ColumnsScript("ProductImagescom.ProductImagescom")]
[BasedOnRow(typeof(ProductImagescomRow), CheckNames = true)]
public class ProductImagescomColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ProductName { get; set; }
    [EditLink]
    public string Image { get; set; }
}